pkg/informers/informers_test.go

```diff
@@ -0,0 +1,264 @@
+	return m, nil
+}
+
+func (m *mockFactory) Namespaces() sets.String {
+	return m.namespaces
+}
+
+func TestInformers(t *testing.T) {
+	t.Run("TestGet", func(t *testing.T) {
+		ifs, err := NewInformersForResource(
+			&mockFactory{
+				namespaces: sets.NewString("foo", "bar"),
+				objects: map[string]runtime.Object{
+					"foo": &monitoringv1.Prometheus{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "foo",
+						},
+					},
+				},
+			},
+			schema.GroupVersionResource{},
+		)
+		if err != nil {
+			t.Error(err)
+			return
+		}
+
+		_, err = ifs.Get("foo")
+		if err != nil {
+			t.Error(err)
+			return
+		}
+
+		_, err = ifs.Get("bar")
+		if !errors.IsNotFound(err) {
+			t.Errorf("expected IsNotFound error, got %v", err)
+			return
+		}
+	})
+}
+
+func TestNewInformerOptions(t *testing.T) {
+	for _, tc := range []struct {
+		name                                string
+		allowedNamespaces, deniedNamespaces map[string]struct{}
+		tweaks                              func(*v1.ListOptions)
+
+		expectedOptions    v1.ListOptions
+		expectedNamespaces []string
+	}{
+		{
+			name:               "all unset",
+			expectedOptions:    v1.ListOptions{},
+			expectedNamespaces: nil,
+		},
+		{
+			name: "allowed namespaces",
+			allowedNamespaces: map[string]struct{}{
+				"foo": {},
+				"bar": {},
+			},
+			expectedOptions: v1.ListOptions{},
+			expectedNamespaces: []string{
+				"foo",
+				"bar",
+			},
+		},
+		{
+			name: "allowed namespaces with a tweak",
+			allowedNamespaces: map[string]struct{}{
+				"foo": {},
+				"bar": {},
+			},
+			tweaks: func(options *v1.ListOptions) {
+				options.FieldSelector = "metadata.name=foo"
+			},
+
+			expectedOptions: v1.ListOptions{
+				FieldSelector: "metadata.name=foo",
+			},
+			expectedNamespaces: []string{
+				"foo",
+				"bar",
+			},
+		},
+		{
+			name: "allowed and ignored denied namespaces",
+			allowedNamespaces: map[string]struct{}{
+				"foo": {},
+				"bar": {},
+			},
+			deniedNamespaces: map[string]struct{}{
+				"denied1": {},
+				"denied2": {},
+			},
+
+			expectedOptions: v1.ListOptions{},
+			expectedNamespaces: []string{
+				"foo",
+				"bar",
+			},
+		},
+		{
+			name: "one allowed namespace and ignored denied namespaces",
+			allowedNamespaces: map[string]struct{}{
+				"foo": {},
+			},
+			deniedNamespaces: map[string]struct{}{
+				"denied1": {},
+				"denied2": {},
+			},
+
+			expectedOptions: v1.ListOptions{},
+			expectedNamespaces: []string{
+				"foo",
+			},
+		},
+		{
+			name: "all allowed namespaces denying namespaces",
+			allowedNamespaces: map[string]struct{}{
+				v1.NamespaceAll: {},
+			},
+			deniedNamespaces: map[string]struct{}{
+				"denied2": {},
+				"denied1": {},
+			},
+
+			expectedNamespaces: []string{
+				v1.NamespaceAll,
+			},
+			expectedOptions: v1.ListOptions{
+				FieldSelector: "metadata.namespace!=denied1,metadata.namespace!=denied2",
+			},
+		},
+		{
+			name: "denied namespaces with tweak",
```

Comment:
I don't see that the `tweaks` field is set.

Reply:
fixed